packages/flutter_tools/lib/src/base/bot_detector.dart

```diff
@@ -5,17 +5,20 @@
 import 'package:meta/meta.dart';
 import 'package:platform/platform.dart';
 
+import '../persistent_tool_state.dart';
 import 'io.dart';
 import 'net.dart';
 
 class BotDetector {
   BotDetector({
     @required HttpClientFactory httpClientFactory,
     @required Platform platform,
+    @required PersistentToolState persistentToolState,
   }) :
     _platform = platform,
     _azureDetector = AzureDetector(
       httpClientFactory: httpClientFactory,
+      persistentToolState: persistentToolState,
```

Comment:
Should this cache `BotDetector.isRunningOnBot` instead of just the `AzureBotDetector` result? That would guard against someone adding another expensive step.

Reply:
If someone does add another bot detector though, we might be incorrectly caching a value for these bots ... through I'm not sure if it matters. I expect that the same bot would not be using flutter over and over...

Reply:
Updated to cache at bot detector level